packages/core/src/controllers/relayer.ts

```diff
@@ -287,7 +287,7 @@ export class Relayer extends IRelayer {
       }
     }
 
-    if (this.hasExperiencedNetworkDisruption || this.connected) {
+    if (this.provider.disconnect && (this.hasExperiencedNetworkDisruption || this.connected)) {
```

Comment:
Would it be safer to use the optional chain operator or it's not needed ?

```suggestion
    if (this.provider?.disconnect && (this.hasExperiencedNetworkDisruption || this.connected)) {
```

Reply:
no need as `this.provider` is `{}` if not yet initialized. Being `{}` is covered by other tests